build/scripts/script.js

```diff
@@ -519,84 +522,137 @@ function updateSearchResults(value) {
+      if (h1[__DEFAULT_SYMBOL__]) {
+        const elem = createResultElement(h1[__DEFAULT_SYMBOL__]);
+        searchResultsElt.appendChild(elem);
       }
-      linkH1.className = "h1";
-      linkH1.textContent = res.item.h1;
-      locationDiv.appendChild(linkH1);
-      needSeparator = true;
-    }
+      for (const [keyH2, valueH2] of Object.entries(h1)) {
+        if (keyH2 === __DEFAULT_SYMBOL__) {
+          continue;
+        }
+        if (valueH2[__DEFAULT_SYMBOL__]) {
+          const elem = createResultElement(valueH2[__DEFAULT_SYMBOL__]);
+          searchResultsElt.appendChild(elem);
+        }
 
-    if (res.item.h2 !== undefined && res.item.h2 !== "") {
-      if (needSeparator) {
-        const separatorSpan = document.createElement("span");
-        separatorSpan.textContent = " > ";
-        locationDiv.appendChild(separatorSpan);
-        needSeparator = false;
-      }
-      let linkH2;
-      if (links.anchorH2 !== undefined) {
-        const href = rootUrl + "/" + links.file + "#" + links.anchorH2;
-        linkH2 = document.createElement("a");
-        linkH2.href = href;
-      } else {
-        linkH2 = document.createElement("span");
+        for (const [keyH3, valueH3] of Object.entries(valueH2)) {
+          if (keyH3 === __DEFAULT_SYMBOL__) {
+            continue;
+          }
+          valueH3.forEach((val) => {
+            const elem = createResultElement(val);
+            searchResultsElt.appendChild(elem);
+          });
+        }
       }
-      linkH2.className = "h2";
-      linkH2.textContent = res.item.h2;
-      locationDiv.appendChild(linkH2);
-      needSeparator = true;
     }
-    if (res.item.h3 !== undefined && res.item.h3 !== "") {
-      if (needSeparator) {
-        const separatorSpan = document.createElement("span");
-        separatorSpan.textContent = " > ";
-        locationDiv.appendChild(separatorSpan);
-        needSeparator = false;
+  }
+}
+
+/**
+ * Add missing top sections headers (h1) in search results.
+ * Search result can be h1, h2 or h3, but the search algorithm can
+ * find a match for a h3 section and not for the h1.
+ * To not display h3 orphan in the tree view, we manually add to
+ * the search result the h1 that is a parent to that h3.
+ * @param {array} searchResults
+ * @returns
+ */
+function addMissingTopSections(searchResults) {
+  const h1Sections = {};
+  const missingH1 = [];
+  for (result of searchResults) {
+    const keyName = `${result.item.file}-${result.item.h1}`;
+    if (!h1Sections[keyName]) {
+      h1Sections[keyName] = [];
+    }
+    h1Sections[keyName].push(result);
+  }
+
+  for (const value of Object.values(h1Sections)) {
+    const topSection = value.find((r) => {
+      return r.item.h2 === undefined;
+    });
+    if (topSection === undefined) {
+      const fakeH1 = structuredClone(value[0]);
+      fakeH1.item.h2 = undefined;
+      fakeH1.item.h3 = undefined;
+      fakeH1.item.body = "";
+      missingH1.push(fakeH1);
```

Comment:
If `topSection === undefined` doesn't it mean that there was no h2 that was `undefined` in which case we still here weirdly add an h1 header without h2 nor h3?

I don't get it

Reply:
I'm not sure I correctly explained but the goal is to always display the full path of the search result is the tree even if the search only has match for a sub item. 

Without adding the top level h1: 
<img width="922" alt="image" src="https://github.com/user-attachments/assets/e86c7ad7-e60c-42d4-a2c7-6bff37dd78e2" />

When adding the top level h1:
<img width="904" alt="image" src="https://github.com/user-attachments/assets/a7e99bce-52e3-4779-b5a0-e419aa1a424e" />


So the goal of this code is to manually create and add the h1 "CLI API" to the search result so it can be displayed.